File: InJava/app/src/main/java/com/barryzea/billingapp/MyApp.java
package com.barryzea.billingapp;

import android.annotation.SuppressLint;
import android.app.Application;

import com.barryzea.billingapp.common.Preferences;

/****
 * Project BillingApp
 * Created by Barry Zea H. on 01/03/23.
 * Copyright (c)  All rights reserved.
 ***/
public class MyApp extends Application {

    @SuppressLint("StaticFieldLeak")
    public  static Preferences prefs;
    @Override
    public void onCreate() {
        super.onCreate();
        prefs = new Preferences(this);

    }


}

File: InKotlin/app/src/main/java/com/barryzea/billingapp/MyApp.kt
package com.barryzea.billingapp

import android.annotation.SuppressLint
import android.app.Application
import com.barryzea.billingapp.common.Preferences

/****
 * Project BillingApp
 * Created by Barry Zea H. on 1/3/23.
 * Copyright (c)  All rights reserved.
 ***/
class MyApp: Application() {
    companion object{
        @SuppressLint("StaticFieldLeak")
        lateinit var prefs: Preferences
    }

    override fun onCreate() {
        super.onCreate()
        prefs = Preferences(applicationContext)
    }
}